src/Translatable/Translatable.php

```diff
@@ -458,4 +459,35 @@ public function __isset($key)
     {
         return $this->isTranslationAttribute($key) || parent::__isset($key);
     }
+
+    public function getTranslationChanges(): array
+    {
+        switch (config('translatable.rule_factory.format')) {
+            case RuleFactory::FORMAT_KEY:
+                $replacementFunc = function (string $key, string $locale): string {
+                    return $key . ':' . $locale;
+                };
+                break;
+            default:
+                $replacementFunc = function (string $key, string $locale): string {
+                    return $locale . '.' . $key;
+                };
+        }
+        $translationChanges = array();
+        foreach ($this->translations as $translation) {
+            $locale = $translation->locale;
+            foreach ($translation->getChanges() as $attribute => $value) {
+                $translationChanges[$replacementFunc($attribute, $locale)] = $value;
```

Comment:
The `Arr::set()` understands the possible dot.notation so that the locale is used as a single array key which will also be possible to fill back into the main model.

```suggestion
                \Illuminate\Support\Arr::set($translationChanges, $replacementFunc($attribute, $locale), $value);
```

Reply:
I understand your intention and –from my first intuition– consider it sexy.
But creating a two-dimensional array would render the format distinction obsolete.
Would the `Arr::set()` method even work with the colon based format?
Depending on the use case it's also questionable whether the language or the translated attribute's name should define the first dimension.

Reply:
The colon format would work as expected.

**colon**
```php
[
  'name:de' => 'Hallo Welt',
  'name:en' => 'Hello World',
]
```

**array**
```php
[
  'de' => [
    'name' => 'Hallo Welt',
  ],
  'en' => [
    'name' => 'Hello World',
  ],
]
```

So in all cases you can take the changes array and fill it in any model - and also use the same display helper you possibly have.

Reply:
So why do we even create different formatted keys if the outcome will be the same?

Reply:
The output isn't the same - but both are usable with the package.